### PR TITLE
Adding HTTPS support for flagd

### PR DESCRIPTION
**Pull request by AlexsJones**
This PR gives the ability to use TLS with flagD as reported https://github.com/open-feature/flagd/issues/97

It introduces two new parameters on the `start` command

```
  -c, --server-cert-path string   server side tls certificate path
  -k, --server-key-path string    server side tls key path
```

These can be pointed to `pem/crt/key` files.

Example is `flagd start -c server-crt.pem -k server-key.pem`

### CODE REVIEW

**Comment by james-milligan** [comment]
Ive tested the grpc connection and it works as expected - unable to connect with insecure dial credentials, but able to connect using the correct certificates 👍

File: pkg/service/http_service.go [hunk 1]
```diff
@@ -29,42 +33,86 @@ type HTTPService struct {
...
+	server := &http.Server{
+		Handler:           mux,
+		ReadHeaderTimeout: 60 * time.Second,
+	}
+	tcpm := cmux.New(l)
+	// We first match on HTTP 1.1 methods.
+	httpl := tcpm.Match(cmux.HTTP1Fast())
+	// If not matched, we assume that its TLS.
+	tlsl := tcpm.Match(cmux.Any())
+	tlsl = s.tlsListener(tlsl)
```
Comment (james-milligan) [inline pkg/service/http_service.go:104]: This will `log.fatal` if no `ServerCertPath` / `ServerKeyPath` are provided.

Comment (AlexsJones) [reply pkg/service/http_service.go:105]: Yeah I just ran into this, it's quite hard to fix

File: pkg/service/http_service.go [hunk 2]
```diff
@@ -29,42 +34,86 @@ type HTTPService struct {
...
+	// Now, we build another mux recursively to match HTTPS and GoRPC.
+	// You can use the same trick for SSH.
+	tlsm := cmux.New(tlsl)
+	httpsl := tlsm.Match(cmux.HTTP1Fast())
+	gorpcl := tlsm.Match(cmux.Any())
 
+	go func() { handleServiceError(server.Serve(httpl)) }()      // HTTP
+	go func() { handleServiceError(server.Serve(httpsl)) }()     // HTTP
```
Comment (james-milligan) [inline pkg/service/http_service.go:113]: I think this should be HTTPS but I may be mistaken

File: .gitignore [hunk 3]
```diff
@@ -1,9 +1,12 @@
+openfeature*
```
Comment (beeme1mr) [inline .gitignore:1]: Why was this added to the gitignore?

File: cmd/start.go [hunk 4]
```diff
@@ -187,6 +193,10 @@ func init() {
 		&syncProvider, "sync-provider", "y", "filepath", "Set a sync provider e.g. filepath or remote")
 	startCmd.Flags().StringVarP(
 		&evaluator, "evaluator", "e", "json", "Set an evaluator e.g. json")
+	startCmd.Flags().StringVarP(
+		&serverCertPath, "server-cert-path", "c", "", "server side tls certificate path")
```
Comment (beeme1mr) [inline cmd/start.go:197]:
```suggestion
		&serverCertPath, "server-cert-path", "c", "", "Server side tls certificate path")
```

File: cmd/start.go [hunk 5]
```diff
@@ -187,6 +193,10 @@ func init() {
...
+	startCmd.Flags().StringVarP(
+		&serverKeyPath, "server-key-path", "k", "", "server side tls key path")
```
Comment (beeme1mr) [inline cmd/start.go:199]:
```suggestion
		&serverKeyPath, "server-key-path", "k", "", "Server side tls key path")
```

File: pkg/service/grpc_service.go [hunk 6]
```diff
@@ -26,9 +28,12 @@ type GRPCService struct {
 	Logger *log.Entry
 }
 
+// Serve allows for the use of GRPC only without HTTPP, where as HTTP service enables both
+// GRPC and http
 func (s *GRPCService) Serve(ctx context.Context, eval eval.IEvaluator) error {
 	s.Eval = eval
 
+	// TODO: Needs TLS implementation
```
Comment (beeme1mr) [inline pkg/service/grpc_service.go:36]: Can you create an issue for this TODO?

**Review by beeme1mr — COMMENTED** [review]
Could you please add an example to the readme?

File: .gitignore (hunk 3 above)
Comment (AlexsJones) [reply .gitignore:1]: This was to avoid certificates being committed, I'll remove it.

File: pkg/service/grpc_service.go (hunk 6 above)
Comment (AlexsJones) [reply pkg/service/grpc_service.go:36]: Done https://github.com/open-feature/flagd/issues/103

File: pkg/service/utils.go [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+		return nil, err
+	}
+
+	config := &tls.Config{
+		Certificates: []tls.Certificate{certificate},
+		Rand:         rand.Reader,
+		MinVersion:   tls.VersionTLS12,
+	}
+
+	return &config.Certificates[0], nil
```
Comment (james-milligan) [inline pkg/service/utils.go:33]: it looks like the returned certificate is being used to reconstruct the `tls.Config` struct in `HTTPService.tlsListener` it may be simpler to just return the config at this point

Comment (AlexsJones) [reply pkg/service/utils.go:33]: Good spot, fixed

**Review by james-milligan — APPROVED** [review]
